dev-packages/application-manager/src/application-package-manager.ts

```diff
@@ -182,7 +182,15 @@ export class ApplicationPackageManager {
             throw new AbortError('Dependencies are out of sync, please run "install" again');
         }
         await ffmpeg.replaceFfmpeg();
-        await ffmpeg.checkFfmpeg();
+        try {
+            await ffmpeg.checkFfmpeg();
+        } catch (error) {
+            if (error.message.includes('(have \'arm64\', need \'x86_64\')') && process.env.ARCH === 'arm64') {
+                console.warn('WARN: ffmpeg checking skipped because mac-o file incompatible in this step');
+            } else {
+                throw error;
+            }
+        }
```

Comment:
Do we still need the try/catch?

The check seemed to fail because the replaced library actually couldn't be used, so it's not just that the check failed, but something actually is wrong with `libffmpeg.dylib`. I would expect a hard error to be thrown in such a case.

Is the issue still fixed even without this specific change?

Reply:
Yes, 
If we can run node that have arm64. No need this try/catch.
But now, this line throws mac-o file error, we can ignore this error for arm64 machine or we can compile fmpeg.node with arm64 too then checking the libffmpeg.dylib by architecture

Reply:
I would prefer finding a solution that allows `checkFfmpeg` to work on `arm64`.

The starting point would be to edit [`@theia/ffmpeg`'s `binding.gyp` file](https://github.com/eclipse-theia/theia/blob/master/dev-packages/ffmpeg/binding.gyp). Not sure what's required to build on M1 yet...

Reply:
To run properly without wrok-around needs compatible node version for arm64. Because, node process 
throws mac-o incompatible file error when processing line like require('...ffmpeg.node' and other native modules(nsfw, vs..))

Reply:
The file I linked in my previous comment is the "Makefile-like" recipe used to build Theia's `ffmpeg.node` file. Please look if you can find how to make it build using the right arch.